mdata/store_cassandra.go

```diff
@@ -435,12 +449,20 @@ func (c *CassandraStore) SearchTable(key, table string, start, end uint32) ([]ch
 	results := make(chan outcome, numQueries)
 	for i := range crrs {
 		crrs[i].out = results
-		c.readQueue <- crrs[i]
+		select {
+		case c.readQueue <- crrs[i]:
+		default:
+			cassReadQueueFull.Inc()
+			return nil, errReadQueueFull
+		}
 	}
 	outcomes := make([]outcome, 0, numQueries)
 
 	seen := 0
 	for o := range results {
+		if o.omitted {
+			return nil, errReadQueueFull
```

Comment:
my suggestion is to return an error that is correct (e.g errReadTooOld or something)